### ISSUE TITLE
NPE when a value of a wrong type is returned

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#29516]
In the following, when I add `books` as the return value, an NPE is thrown.
```ballerina
resource function get books(int? 'limit) returns Books|Error {
        stream<record {| anydata...; |}, sql:Error> result = db->query("SELECT * from books");
        Book[] books = [];
        sql:Error? forEach = result.forEach(function (record{} row) {
            io:println(row);
            books.push(<Book>row);
        });
        return books;
    }

type Book record {
    int id;
    string name;
    string author;
};

type Books record {
    Book[] bookslist;
};

type Error record {
    int code;
    string message;
};
```
The error:
```
[Error - 10:43:56 PM] CodeAction 'FixReturnTypeCodeAction' failed! {, error: 'java.lang.NullPointerException'}
java.lang.NullPointerException
	at org.ballerinalang.langserver.codeaction.providers.changetype.FixReturnTypeCodeAction.getFunctionNode(FixReturnTypeCodeAction.java:133)
	at org.ballerinalang.langserver.codeaction.providers.changetype.FixReturnTypeCodeAction.getDiagBasedCodeActions(FixReturnTypeCodeAction.java:75)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$2(CodeActionRouter.java:96)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.lambda$getAvailableCodeActions$3(CodeActionRouter.java:94)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:92)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:133)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:335)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by IMS94** [comment]
@pubudu91 this is related to #29314. And is fixed by #29337. Will be available in the next release.

**Comment by nadeeshaan** [comment]
Closing based on the above comment
